app/assets/javascripts/discourse/app/services/document-title.js

```diff
@@ -0,0 +1,105 @@
+import Service from "@ember/service";
+import { inject as service } from "@ember/service";
+import getURL from "discourse-common/lib/get-url";
+
+export default Service.extend({
+  appEvents: service(),
+  contextCount: null,
+  notificationCount: null,
+  _title: null,
+  _backgroundNotify: null,
+
+  init() {
+    this._super(...arguments);
+    this.reset();
+  },
+
+  reset() {
+    this.contextCount = 0;
+    this.notificationCount = 0;
+    this._title = null;
+    this._backgroundNotify = null;
+  },
+
+  getTitle() {
+    return this._title;
+  },
+
+  setTitle(title) {
+    this._title = title;
+    this._renderTitle();
+  },
+
+  updateFocus(focus) {
+    let { session } = this;
+
+    session.hasFocus = focus;
+
+    if (session.hasFocus && this._backgroundNotify) {
+      this.updateContextCount(0);
+    }
+    this._backgroundNotify = false;
+
+    if (session.hasFocus) {
+      this.notificationCount = 0;
+    }
+    this.appEvents.trigger("discourse:focus-changed", session.hasFocus);
+    this._renderTitle();
+  },
+
+  updateContextCount(count) {
+    this.contextCount = count;
+    this._renderTitle();
+  },
+
+  updateNotificationCount(count) {
+    if (!this.session.hasFocus) {
+      this.notificationCount = count;
+      this._renderFavicon();
+      this._renderTitle();
+    }
+  },
+
+  incrementBackgroundContextCount() {
+    if (!this.session.hasFocus) {
+      this._backgroundNotify = true;
+      this.contextCount += 1;
+      this._renderFavicon();
+      this._renderTitle();
+    }
+  },
+
+  _displayCount() {
+    return this.currentUser &&
+      this.currentUser.get("title_count_mode") === "notifications"
+      ? this.notificationCount
+      : this.contextCount;
+  },
+
+  _renderTitle() {
+    let title = this._title || this.siteSettings.title;
+
+    let displayCount = this._displayCount();
+    let dynamicFavicon = this.currentUser && this.currentUser.dynamic_favicon;
+    if (displayCount > 0 && !dynamicFavicon) {
+      title = `(${displayCount}) ${title}`;
+    }
+
+    document.title = title;
+  },
+
+  _renderFavicon() {
+    if (this.currentUser && this.currentUser.dynamic_favicon) {
+      let url = this.siteSettings.site_favicon_url;
+
+      // Since the favicon is cached on the browser for a really long time, we
+      // append the favicon_url as query params to the path so that the cache
+      // is not used when the favicon changes.
+      if (/^http/.test(url)) {
+        url = getURL("/favicon/proxied?" + encodeURIComponent(url));
+      }
+
+      new window.Favcount(url).set(this._displayCount());
```

Comment:
does it pass linting ? I thought eslint wouldn't accept to use new without assignment.

Reply:
Hmmm I believe it did (that code is moved from elsewhere). I'll confirm shortly.